### ISSUE TITLE
Feature request: Return uploads in order

### ISSUE THREAD

**Issue by mymikemiller** [Hexer10/youtube_explode_dart#40]
**Is your feature request related to a problem? Please describe.**
Problem: Uploads are returned in reverse order (newer videos first, older videos last). I need them in the order the user uploaded them in, which currently requires me to fetch the entire list so it can be reversed before doing anything with it.

**Describe the solution you'd like**
Would be nice to be able to obtain a Stream of uploads in the order they were uploaded, like if you selected "Sort By: Date added (oldest)" on the channel's upload page.

**Describe alternatives you've considered**
Use .toList().reversed on the Stream. This defeats the purpose of using Streams to avoid unnecessary computation and http requests. The entire list of the user's uploads must be fetched before I can start work on any of them (my work must be performed on the older videos first)

**How should this be implemented**
Maybe something can be added to the [playerlist_response](https://youtube.com/list_ajax?style=json&action_get_list=1&list=$id&index=$index&hl=en) request to get the results to come back in reverse order. I'm not sure how the channel's videos page does it, when the results are sorted oldest to newest.

**Comment by Hexer10** [comment]
So, as of now the videos are not fetched directly from the channel's YouTube page, but from a playlist containing all the channel videos, and I think there is no way to reverse that playlist.
Parsing from the videos page ideally should be too hard (I haven't looked into this yet), doing so would allow to sort from the newest, oldest and popularity...
I'll see in the next days what I can do

**Comment by mymikemiller** [comment]
Thanks, @Hexer10! Let me know if I can help in any way. Adding this functionality would make your Dart library differ from the .NET library - that's something you're ok with? Or should this be a feature request on [Youtube Explode](https://github.com/Tyrrrz/YoutubeExplode) first?

I'm loving this library, by the way. Great work!

**Comment by Hexer10** [comment]
I have no issues in differing from the original library, in facts I already did it implementing a comments API.
If you want to contribute in any way feel free to do it, I'd really appreciate it!

**Comment by mymikemiller** [comment]
This feature may be a bit above my expertise, but I'm happy to contribute where I can!

**Comment by Hexer10** [comment]
Don't worry, I already know how this should be made, its just a bit tedious to go thru the youtube page and reverse engineer the youtube channel page since it has so many variables with every channel (See for example between official music channel and "normal" channel).
For that reason it may take a while, but it is on my todo list.

**Comment by Hexer10** [comment]
I'm sorry this took a while but now it should be possible to fetch the videos by popularity and publish date here is an example:
```
  var yt = YoutubeExplode();
  var up = await yt.channels
      .getUploadsFromPage('UCVFAPt7jWhmsZZl9DhxSfXQ', VideoSorting.newest) //Also VideoSorting.oldest, VideoSorting.popularity
      .take(30)
      .toList();
  print(up);
```


Make sure to limit the result using `.take` or it may end up taking ages to finish the request on channels with lots of videos.

**Comment by mymikemiller** [comment]
I tried this out and it works great, although for some reason videoTitle isn't being populated.

```
  final channelId = ChannelId('UCNl_4FD4qQdZZJMzAM7LJqQ');
  final stream = yt.channels.getUploadsFromPage(channelId, VideoSorting.oldest);
  await for (var video in stream) {
    print('${video.videoId}: ${video.videoTitle}');
  }
```

Output:
```
ufD8H8aKDvU: null
hgu6XohHdU0: null
7wpZWBWO9N0: null
...
```

I realized that this feature actually won't make my life any easier unless there's a way to specify what date, video or page to start looking from. I'd like to retrieve only the videos published after a specific date, in the order they were published (aside: is it possible to get the published date from your library? All I can see is uploadDate, but that doesn't match the order they come back in). It just changes the problem. I used to get the whole list backward until hitting the desired date, then reverse it. Now I can avoid reversing the list, but I have to fetch all the pages up until the date I want to start at, throw all that away, then start building my list.

The old way is better for my needs, especially when creators have tons of uploads.

Maybe if I could the cache `channel_upload_page::continuation` variable from the previous run, I could start my next run there and pick up where I left off, which is my ultimate goal. What I have now is working fine, though. Thanks for putting in the time to add this feature!
